api/envoy/extensions/filters/http/grpc_json_transcoder/v3/transcoder.proto

```diff
@@ -288,4 +288,17 @@ message GrpcJsonTranscoder {
   //
   // If unset, the current stream buffer size is used.
   google.protobuf.UInt32Value max_response_body_size = 16 [(validate.rules).uint32 = {gt: 0}];
+
+  // If true, query parameters that cannot be mapped to a corresponding
+  // protobuf field are captured in an HttpBody extension of UnknownVariableBindings.
+  // This overrides ``ignore_unknown_query_parameters`` and ``reject_unknown_query_parameters``.
+  bool capture_unknown_query_parameters = 17;
+}
+
+// ``UnknownVariableBindings`` is added as an extension field in ``HttpBody`` if
+// ``GrpcJsonTranscoder::capture_unknown_query_parameters`` is true and unknown query
+// parameters were present in the request.
+message UnknownVariableBindings {
```

Comment:
ooc why bindings rather than UnknownQueryParams or something?

Reply:
Because renaming it would be a massive deployment pain for us, as this message is all throughout a whole bunch of client code already. :D

Reply:
On second thought, maybe it makes more sense to bind this with a different extension with a better API, have the patch and the new one both operating during the transition with the client preferring the new encoding, and that way I can do an API that isn't as bad as the original one.